modules/calendar/calendar.py

```diff
@@ -30,29 +30,35 @@ def from_dict(cls, details: Dict[str, str]) -> "Calendar":
 
 	@classmethod
 	def get_calendar(
-		cls, ctx: SlashContext, groups: Iterable[Group], group_id: int = None
+		cls,
+		ctx: SlashContext,
+		groups: Optional[Iterable[Group]] = None,
```

Comment:
Why does this need to be optional if `self.groups` is always passed as a parameter?

And if it could be skipped, why not just remove the parameter and have it always fetched within the function?

Reply:
It takes it as a parameter for efficiency, but if one were to skip it it's still computable, so I just added that in because I could